lib/ioki/webhooks/event.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require 'ostruct'
+
+module Ioki
+  module Webhooks
+    class Event < OpenStruct
+
+      def model
+        model_class.new data
+      end
+
+      def model_class
+        Ioki::Model::Webhooks.const_get model_class_name
+      end
+
+      def model_class_name
+        camelize data['type']
+      end
+
+      def action
+        event_type&.split('.')&.last
+      end
+
+      private
+
+      def camelize(term)
```

Comment:
I'm wondering if it would make sense to copy the Rails' implementation as that's what I'd expect most people using the gem are familiar with. Otherwise there might be subtle differences.

See https://apidock.com/rails/v6.1.3.1/String/camelize. Rails 7's is harder to read in my opinion: https://github.com/rails/rails/blob/de53ba56cab69fb9707785a397a59ac4aaee9d6f/activesupport/lib/active_support/inflector/methods.rb#L96.

Maybe it would also be nice to extract the function to a module as `camelize` is a concept that makes sense for all parts of the app.

What do you think @eikes?

Reply:
It is in fact copied from rails 7 here: https://github.com/rails/rails/blob/de53ba56cab69fb9707785a397a59ac4aaee9d6f/activesupport/lib/active_support/inflector/methods.rb#L69

But I had to remove everything related to inflections. Finally I had to replace the $1 and $2, because rubocop didn't like that.

In general though, I'd prefer to use ActiveSupport, but I was told not to. What are your thoughts on using integrating ActiveSupport? It would let us remove some reimplementations like this and everyone uses it anyways...

Reply:
Huh. I looked at that method as well and didn't recognise it. Odd. I'd personally be Ok with ActiveSupport, but was also told not to use it and I'd stick with that. I think the next best option would be to have a module for "the stuff copied from Rails" with some comments and links to the original source.

Reply:
OK, let's create a Ioki::Support module after merging this!